tests/estimators/speech_recognition/test_tensorflow_lingvo.py

```diff
@@ -0,0 +1,329 @@
+        lingvo = LingvoAsr()
+        features_tf = lingvo._create_log_mel_features(lingvo._x_padded)
+
+        features = lingvo._sess.run(features_tf, {lingvo._x_padded: test_input})
+        assert features.shape[2] == 80
+        assert len(features.shape) == 4
+
+    def test_pad_audio_input(self):
+        import tensorflow.compat.v1 as tf1
+
+        tf1.reset_default_graph()
+
+        test_input = np.array([np.array([1]), np.array([2] * 480)], dtype=object)
+        test_mask = np.array([[True] + [False] * 479, [True] * 480])
+        test_output = np.array([[1] + [0] * 479, [2] * 480])
+
+        lingvo = LingvoAsr()
+        output, mask, mask_freq = lingvo._pad_audio_input(test_input)
+        assert_array_equal(test_output, output)
+        assert_array_equal(test_mask, mask)
+
+    def test_predict_batch(self, audio_batch_padded):
+        import tensorflow.compat.v1 as tf1
+
+        tf1.reset_default_graph()
+
+        test_input, test_mask_frequency = audio_batch_padded
+        test_target_dummy = np.array(["DUMMY"] * test_input.shape[0])
+
+        lingvo = LingvoAsr()
+        feed_dict = {
+            lingvo._x_padded: test_input,
+            lingvo._y_target: test_target_dummy,
+            lingvo._mask_frequency: test_mask_frequency,
+        }
+        predictions = lingvo._sess.run(lingvo._predict_batch_op, feed_dict)
+        assert set(predictions.keys()).issuperset(
+            {
+                "target_ids",
+                "target_labels",
+                "target_weights",
+                "target_paddings",
+                "transcripts",
+                "topk_decoded",
+                "topk_ids",
+                "topk_lens",
+                "topk_scores",
+                "norm_wer_errors",
+                "norm_wer_words",
+                "utt_id",
+            }
+        )
+
+    def test_predict(self, audio_data):
+        import tensorflow.compat.v1 as tf1
+
+        tf1.reset_default_graph()
+
+        test_input = audio_data
+
+        lingvo = LingvoAsr()
+        predictions = lingvo.predict(test_input, batch_size=2)
+        assert predictions.shape[0] == test_input.shape[0]
+        assert isinstance(predictions[0], np.str_)
+
+    def test_loss_gradient_tensor(self, audio_batch_padded):
+        import tensorflow.compat.v1 as tf1
+
+        tf1.reset_default_graph()
+
+        test_input, test_mask_frequency = audio_batch_padded
+        test_target_dummy = np.array(["DUMMY"] * test_input.shape[0])
+
+        lingvo = LingvoAsr()
+        feed_dict = {
+            lingvo._x_padded: test_input,
+            lingvo._y_target: test_target_dummy,
+            lingvo._mask_frequency: test_mask_frequency,
+        }
+        loss_gradient = lingvo._sess.run(lingvo._loss_gradient_op, feed_dict)
+        assert test_input.shape == loss_gradient.shape
+        assert loss_gradient.sum() == 0.0
+
+    def test_loss_gradient_per_batch(self, audio_data):
+        import tensorflow.compat.v1 as tf1
```

Comment:
aren't `test_loss_gradient_per_batch` and `test_loss_gradient_per_sequence` duplicated tests? I'm trying to see how do they differ but can't see any difference. 
In general, there's 2 conventions that would apply here. 
Option 1: If the only thing that differs from the tests is an input parameter, then both tests should be merged into one and instead the `@pytest.mark.parametrize` should be used (see [test_fast_gradient.py/test_norm_images](https://github.com/Trusted-AI/adversarial-robustness-toolbox/blob/main/tests/attacks/evasion/test_fast_gradient.py) as an example)

Option 2: if there's many differences between the two but still quite a lot of code duplicated between them then we need to refactor the identical code into a separate method called by both tests named `back_end_my_function_etc` (see section 3.3.3 in [wiki doc](https://github.com/Trusted-AI/adversarial-robustness-toolbox/wiki/ART-Unit-Testing))